CHANGES/1039.bugfix.rst

```diff
@@ -0,0 +1,25 @@
+:meth:`URL.join() <yarl.URL.join>` has been changed to match
+:rfc:`3986` and align with
+:meth:`/ operation <yarl.URL.__truediv__>` and :meth:`URL.joinpath() <yarl.URL.joinpath>`
+when joining URLs with empty segments.
+Previously :py:func:`urllib.parse.urljoin` was used,
+which has known issues with empty segments
+(`python/cpython#84774 <https://github.com/python/cpython/issues/84774>`_).
+
+Due to the semantics of :meth:`URL.join() <yarl.URL.join>`, joining an
+URL with scheme requires making it relative, prefixing with ``./``.
+
+.. code-block:: python
+
+   URL("https://web.archive.org/web/").join(URL("./https://github.com/aio-libs/yarl"))
```

Comment:
Just got an idea: what if we convert the code blocks here from `python` to `pycon` (as in “python console”: https://pygments.org/docs/lexers/#pygments.lexers.python.PythonConsoleLexer). The syntax would need to match the default REPL. So the first line would be prefixed with `>>> ` and the following one would be the output. This would get highlighted appropriately and demonstrate better what result is expected.
This could later be used as a doctest (if we enable it in sphinx).

Reply:
testing

Reply:
added the expected result

Reply:
thanks! it's much more visual now!